packages/react-components/src/components/table-output-component.tsx

```diff
@@ -466,29 +463,21 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
         }
     }
 
-    private handleRowSelectionChange(load?: any | undefined) {
+    private selectAndNavigateToState(load?: any | undefined) {
+        const startTimestamp = this.startTimestamp;
+        const endTimestamp = this.endTimestamp;
         if (this.timestampCol) {
-            const startTimestamp = String(this.startTimestamp);
-            const endTimestamp = String(this.endTimestamp);
-            const payload = { startTimestamp, endTimestamp, load };
-            this.prevStartTimestamp = BigInt(startTimestamp);
-            this.eventSignal = true;
-            this.handleSelectionRangeUpdate(payload);
-            signalManager().fireSelectionChangedSignal(payload);
-        }
-    }
-
-    private handleSelectionRangeUpdate(payload: { [key: string]: string; }) {
-        const offset = this.props.viewRange.getOffset() || BigInt(0);
-        const startTimestamp = payload['startTimestamp'];
-        const endTimestamp = payload['endTimestamp'];
-        if (startTimestamp !== undefined && endTimestamp !== undefined) {
-            const selectionRangeStart = BigInt(startTimestamp) - offset;
-            const selectionRangeEnd = BigInt(endTimestamp) - offset;
-            this.props.unitController.selectionRange = {
-                start: selectionRangeStart,
-                end: selectionRangeEnd
-            };
+            const offset = this.props.viewRange.getOffset() || BigInt(0);
+            const payload = {
+                id: this.props.traceId,
+                timeRange: {
+                    start: startTimestamp - offset,
+                    end: endTimestamp - offset
+                }
+            } as TimeRangeUpdatePayload;
+            // Fire signals to center the state and change the selection range
+            signalManager().fireRequestSelectionRangeChange(payload);
+            signalManager().fireItemPropertiesSignalUpdated({ startTimestamp, endTimestamp, ...load });
```

Comment:
I see what you mean, the item properties and selection change signal are very similar. However, the item properties are used to show the data to the user (Items Properties view). The selection change signal is more an internal (non-user facing) signal and used sync views.

Adding startTimestamp and endTimestamp to the items properties the items properties view when selecting a n event in the table, and startTimestamp/endTimestamp has no meaning for the selected event. I would keep both signals separate.